srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Hooks/WooCommerce/FieldElements.php

```diff
@@ -46,17 +48,28 @@ public static function charList($username, $deleted = false): void {
                         $bannedChars[] = $value->getName();
                         continue;
                     }
-                    echo '<option value="' . $value->getGuid() . '">' . $value->getName() . '</option>';
+                    echo '<option value="' . $value->getGuid() . '" data-charicon="' . ACORE_URL_PLG . "web/assets/race/" . $value->getRace() . ($value->getGender() == 0 ? "m" : "f") . ".webp" . '" data-classicon="' . ACORE_URL_PLG . "web/assets/class/" . $value->getClass() . ".webp" . '">' . $value->getName() . '</option>';
                 endforeach;
             } else {
                 foreach ($deletedCharacters as $key => $value):
-                    echo '<option value="' . $value->getGuid() . '">' . $value->getDeletedName() . '</option>';
+                    echo '<option value="' . $value->getGuid() . '" data-charicon="' . ACORE_URL_PLG . "web/assets/race/" . $value->getRace() . ($value->getGender() == 0 ? "m" : "f") . ".webp" . '" data-classicon="' . ACORE_URL_PLG . "web/assets/class/" . $value->getClass() . ".webp" . '">' . $value->getDeletedName() . '</option>';
```

Comment:
save these two URLs in two constant variables to avoid code redundancy, like $charRaceIcon and $charClassIcon

Reply:
Sorry, getting a bit stuck on this trying to figure it out. Save the variables at the start of the file or in the foreach loop? If it's at the start of the file with the other variables I need to find a working solution since I haven't found that yet.

Reply:
you're right, make a function for it like, `getRaceUrl()` and `getClassUrl()`